### PR TITLE
Add API validation for infrastructure configuration

### LINKED ISSUE #110: Add API validation for infrastructure configuration
# Summary

Implement `ValidateInfrastructureConfig` and `ValidateInfrastructureConfigUpdate` in the `apis` package.

## Definition of done

- Add validation logic for create and update
- Ensure correct behaviour by providing unit tests

### PR DESCRIPTION
**Pull request by kasabe28**
# Proposed Changes

- Add validation logic for create and update
- Ensure correct behavior by providing unit tests

Fixes #110

### CODE REVIEW

File: pkg/apis/onmetal/validation/infrastructure_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		}
+	})
+
+	Describe("#ValidateInfrastructureConfig", func() {
+		It("should return no errors for a valid configuration", func() {
+			Expect(ValidateInfrastructureConfig(infra, pointer.String(""), pointer.String(""), pointer.String(""), fldPath)).To(BeEmpty())
+		})
+
+		It("should fail with invalid network reference", func() {
+			infra.NetworkRef.Name = ""
```
Comment (afritzler) [inline pkg/apis/onmetal/validation/infrastructure_test.go:50]: I would use here something like "foo%" as an invalid network name.

File: pkg/apis/onmetal/validation/infrastructure_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		infra = &apisonmetal.InfrastructureConfig{
+			NetworkRef: &corev1.LocalObjectReference{
+				Name: networkName,
+			},
+		}
+	})
+
+	Describe("#ValidateInfrastructureConfig", func() {
+		It("should return no errors for a valid configuration", func() {
+			Expect(ValidateInfrastructureConfig(infra, pointer.String(""), pointer.String(""), pointer.String(""), fldPath)).To(BeEmpty())
```
Comment (afritzler) [inline pkg/apis/onmetal/validation/infrastructure_test.go:46]: Just pass on `nil` for the empty strings.

File: pkg/apis/onmetal/validation/infrastructure_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		})
+
+		It("should fail with invalid network reference", func() {
+			infra.NetworkRef.Name = ""
+
+			errorList := ValidateInfrastructureConfig(infra, pointer.String(""), pointer.String(""), pointer.String(""), fldPath)
```
Comment (afritzler) [inline pkg/apis/onmetal/validation/infrastructure_test.go:52]: Same as above.

**Review by afritzler — CHANGES_REQUESTED** [review]
Thanks a lot @kasabe28 for this great PR. Apart from some minor things it looks good.

File: pkg/apis/onmetal/validation/infrastructure_test.go (hunk 1 above)
Comment (kasabe28) [reply pkg/apis/onmetal/validation/infrastructure_test.go]: Used "my%network" as an invalid network name

File: pkg/apis/onmetal/validation/infrastructure_test.go (hunk 2 above)
Comment (kasabe28) [reply pkg/apis/onmetal/validation/infrastructure_test.go]: Done

File: pkg/apis/onmetal/validation/infrastructure_test.go (hunk 3 above)
Comment (kasabe28) [reply pkg/apis/onmetal/validation/infrastructure_test.go]: Done

**Review by afritzler — APPROVED** [review]

**Pull request merged**